src/intro-to-c/conditionals/conditionals.md

```diff
@@ -0,0 +1,284 @@
+
+	return 0;
+}
+```
+
+## Multi-condition Statements
+
+---
+
+`if` statements and the ternary operator can actually handle multiple comparisons or conditions. Looking back at the example in the `else if` section, it can be more efficiently rewritten like this:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	if(sunny == 1 || dark == 1)
+	{
+		close_the_blinds();
+	}
+	else if(snowy == 1 || rainy == 1)
+	{
+		keep_the_blinds_open();
+	}
+	else
+	{
+		go_for_a_drive();
+	}
+	return 0;
+}
+```
+
+Let's try it with the ternary operator.
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int m = 20, n = 15, g = 17, p = 3;
+
+	(m > n && m > g && m > p) ? printf("%d is the greatest.\n", m) : printf("%d is not the greatest.\n", m);
+
+return 0;
+}
+```
+
+Order doesn't technically matter, but the best practice is to order them in a way that allows for potential short circuiting.
+
+## Short Circuiting
+
+---
+
+When writing conditional statements, it is good practice to write them in a way that allows for the compiler to skip the evaluation of further conditions when possible. This skip by the compiler is called short circuiting. Short circuiting is possible when evaluating logical `and` and logical `or`. For example, if the left side of a logical and is false, the whole expression will be false regardless of the right side of the expression, so the compiler skips this evaluation. Here is an example of such a case:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int x = 1, y = 2, z = 2;
+
+	if(x == y && y == z)
+	{
+		printf("Everything is the same!\n");
+	}
+	else
+	{
+		printf("Everything is NOT the same!\n");
+	}
+	return 0;
+}
+```
+
+`x` and `y` are not equivalent, so the comparison would resolve to false, rendering the whole statement false.
+
+With a slight change, here is a similar example for a logical or short circuit:
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int x = 1, y = 2, z = 2;
+
+	if(y == z || x == y)
+	{
+		printf("Something is the same!\n");
+	}
+	else
+	{
+		printf("Nothing is the same!\n");
+	}
+	return 0;
+}
+```
+
+Note that logical or will resolved if **either** side of the expression is true, so comparing `y` and `z` will resolve to true, rendering the whole statement true.
+
+Here is a small program that will help in visualizing the short circuit. Feel free to copy this and plug it into a compiler!
+
+``` c
+#include <stdio.h>
+
+int main(void)
+{
+	int x = 1, y = 2, z = 2;
+
+	if(y == z || x++)
+	{
+		printf("Something is the same!\n");
+	}
+	else
+	{
+		printf("Nothing is the same!\n");
+	}
+	printf("x = %d\n", x);
+	return 0;
+}
+```
+
+When you run this program, the output will show that `x = 1`, even though it was incremented in the in the if statement. This part of the code was actually never executed due to the short circuit, so `x` was never incremented. This code can be slightly manipulated to show the short circuiting of the logical `and` as well, but I will leave that for you to try.
```

Comment:
`x` was not actually incremented so I was confused by the first sentence until I read the next sentence. This just adds a little hint for the reader.

```suggestion
When you run this program, the output will show that `x = 1`, even though it was (supposedly) incremented in the in the if statement. This part of the code was actually never executed due to the short circuit, so `x` was never incremented. This code can be slightly manipulated to show the short circuiting of the logical `and` as well, but I will leave that for you to try.
```